src/App.tsx

```diff
@@ -1,24 +1,159 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { ErrorType } from './types/ErrorType';
+import { FilterBy } from './types/FilterBy';
 
-const USER_ID = 0;
+const USER_ID = 6481;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState(ErrorType.None);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isInputActive, setIsInputActive] = useState(true);
+
+  const [updatingId, setUpdatingId] = useState([0]);
+
+  const getTodosFromServer = async () => {
+    try {
+      setErrorMessage(ErrorType.None);
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setErrorMessage(ErrorType.Load);
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const allTodosCount = todos.length;
+
+  const activeTodosCount = useMemo(
+    () => todos.filter(todo => !todo.completed).length,
+    [todos],
+  );
+
+  useEffect(() => {
+    const filteredTodos = (() => {
+      switch (filterBy) {
+        case FilterBy.All:
+          return todos;
+        case FilterBy.Active:
+          return todos.filter(todo => !todo.completed);
+        case FilterBy.Completed:
+          return todos.filter(todo => todo.completed);
+        default:
+          return todos;
+      }
+    });
+
+    setVisibleTodos(filteredTodos);
+  }, [filterBy, todos]);
+
+  const addTodo = (title: string) => {
+    if (!title.trim()) {
+      setErrorMessage(ErrorType.Title);
+
+      return;
+    }
+
+    const createNewTodo = async () => {
+      try {
+        setErrorMessage(ErrorType.None);
+        const newTodo = {
+          userId: USER_ID,
+          title,
+          completed: false,
+        };
+
+        setIsInputActive(false);
+        setTempTodo({ ...newTodo, id: 0 });
+        await createTodo(USER_ID, newTodo);
```

Comment:
this method returns instance of created todo which have id. So write result to variable and use for function `setTodos` - add created todo to your array.
And explain please for what you need `tempTodo`?

Reply:
TempTodo is a part of task) After clicking `enter`, there may be a delay in creating a _todo_, but it is important for user to see that the _todo_ is being created. That's why, a _tempTodo_ with loader is displayed at the end of the list, and after creating a "permanent" _todo_, the  _tempTodo_ is deleted.